src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommand.cs

```diff
@@ -0,0 +1,165 @@
+
+                var includeDeprecated = listpkg.Option(
+                    "--deprecated",
+                    Strings.ListPkg_DeprecatedDescription,
+                    CommandOptionType.NoValue);
+
+                var includeTransitive = listpkg.Option(
+                    "--include-transitive",
+                    Strings.ListPkg_TransitiveDescription,
+                    CommandOptionType.NoValue);
+
+                var prerelease = listpkg.Option(
+                    "--include-prerelease",
+                    Strings.ListPkg_PrereleaseDescription,
+                    CommandOptionType.NoValue);
+
+                var highestPatch = listpkg.Option(
+                    "--highest-patch",
+                    Strings.ListPkg_HighestPatchDescription,
+                    CommandOptionType.NoValue);
+
+                var highestMinor = listpkg.Option(
+                    "--highest-minor",
+                    Strings.ListPkg_HighestMinorDescription,
+                    CommandOptionType.NoValue);
+
+                var source = listpkg.Option(
+                    "--source",
+                    Strings.ListPkg_SourceDescription,
+                    CommandOptionType.MultipleValue);
+
+                var config = listpkg.Option(
+                    "--config",
+                    Strings.ListPkg_ConfigDescription,
+                    CommandOptionType.SingleValue);
+
+                listpkg.OnExecute(async () =>
+                {
+                    var logger = getLogger();
+
+                    var settings = ProcessConfigFile(config.Value(), path.Value);
+                    var sources = source.Values;
+                    
+                    var packageSources = GetPackageSources(settings, sources, config);
+
+                    VerifyValidFrameworks(framework);
+                    
+                    var packageRefArgs = new ListPackageArgs(
+                        path.Value,
+                        packageSources,
+                        framework.Values,
+                        includeOutdated.HasValue(),
+                        includeDeprecated.HasValue(),
+                        includeTransitive.HasValue(),
+                        prerelease.HasValue(),
+                        highestPatch.HasValue(),
+                        highestMinor.HasValue(),
+                        logger,
+                        CancellationToken.None);
+
+                    var listPackageCommandRunner = getCommandRunner();
+                    await listPackageCommandRunner.ExecuteCommandAsync(packageRefArgs);
+                    return 0;
+                });
+            });
+        }
+
+        private static void VerifyValidFrameworks(CommandOption framework)
+        {
+            var frameworks = framework.Values.Select(f =>
+                                NuGetFramework.Parse(f.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries).Select(s => s.Trim()).ToArray()[0]));
```

Comment:
Use TryParse instead of Parse. Parse directly throws an exception as far as I remember.

Reply:
Parse does not throw, which is why I have the line right after